kernel/syscall.c

```diff
@@ -2656,6 +2656,84 @@ long myst_syscall_recvmsg(int sockfd, struct msghdr* msg, int flags)
     return ret;
 }
 
+long myst_syscall_sendmmsg(
+    int sockfd,
+    struct mmsghdr* msgvec,
+    unsigned int vlen,
+    int flags)
+{
+    long ret = 0;
+    myst_fdtable_t* fdtable = myst_fdtable_current();
+    myst_sockdev_t* sd;
+    myst_sock_t* sock;
+    unsigned int cnt;
+
+    ECHECK(myst_fdtable_get_sock(fdtable, sockfd, &sd, &sock));
+    if (!vlen)
+        return 0;
+    for (cnt = 0; cnt < vlen; cnt++)
+    {
+        ret = (*sd->sd_sendmsg)(sd, sock, &msgvec[cnt].msg_hdr, flags);
+        if (ret < 0)
+            break;
+        msgvec[cnt].msg_len = ret;
+    }
+    // Only return err when zero msg was sent
+    ret = cnt ? (long)cnt : ret;
+
+done:
+    return ret;
+}
+
+long myst_syscall_recvmmsg(
+    int sockfd,
+    struct mmsghdr* msgvec,
+    unsigned int vlen,
+    int flags,
+    struct timespec* timeout)
+{
+    long ret = 0;
+    myst_fdtable_t* fdtable = myst_fdtable_current();
+    myst_sockdev_t* sd;
+    myst_sock_t* sock;
+    struct timespec start;
+    struct timespec now;
+    long expire;
+    unsigned int cnt;
+
```

Comment:
Missing check against msgvec

Reply:
Same here. When `vlen == 0`, msgvec can be NULL.